### PR TITLE
Python modernize codejail

### PR DESCRIPTION
**Pull request by cclauss**
Replaces #65 Related to #64

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @cclauss! I've created [OSPR-3228](https://openedx.atlassian.net/browse/OSPR-3228) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@cclauss Thank you for your contribution. @nedbat Can you give this a look?

File: codejail/django_integration.py [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
 
+from __future__ import absolute_import
 from django.conf import settings as django_settings
 from django.core.exceptions import MiddlewareNotUsed
 
 from .integration import configure_from_settings
 
 
-class ConfigureCodeJailMiddleware(object):
+class ConfigureCodeJailMiddleware():
```
Comment (nedbat) [inline codejail/django_integration.py]: I would rather you didn't remove object as a base class while we are still supporting Python 2.

File: codejail/jail.py [hunk 2]
```diff
@@ -91,12 +92,8 @@ def __getitem__(self, item):
...
         For refactoring ease, mimic the old COMMANDS[x] dict
         """
-        if item == 'user':
-            return self.user
-        elif item == 'cmdline_start':
-            return self.cmdline_start
-        else:
-            raise KeyError(item)
+        return {'user': self.user,
+                'cmdline_start': self.cmdline_start}[item]
```
Comment (nedbat) [inline codejail/jail.py]: I prefer the spelled-out if/elif structure.

File: codejail/proxy.py [hunk 3]
```diff
@@ -17,8 +18,8 @@
 # communicate a few values, and unpack them.  Lastly, we need to be sure we can
 # handle binary data.  Serializing with repr() and deserializing the literals
 # that result give us all the properties we need.
-serialize = repr
-deserialize = ast.literal_eval
+SERIALIZE = repr
+DESERIALIZE = ast.literal_eval
```
Comment (nedbat) [inline codejail/proxy.py]: All functions are constants, I don't see a reason to spell them as if they were numeric constant.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @cclauss! I've created [OSPR-3471](https://openedx.atlassian.net/browse/OSPR-3471) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by edx-webhook** [comment]
@cclauss Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/codejail&entry.867055334=https://github.com/edx/codejail/pull/66&entry.1484655318=https://github.com/cclauss&entry.752974735=2019-03-19+22:35&entry.1917517419=2019-05-21+22:02&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**